azurerm/resource_arm_mssql_elasticpool.go

```diff
@@ -0,0 +1,438 @@
+					return fmt.Errorf("BusinessCritical pricing tier only supports upto 80 vCores")
+				}
+
+				if capacity.(int) < 2 {
+					return fmt.Errorf("BusinessCritical pricing tier must have a minimum of 2 vCores")
+				}
+
+				switch {
+				case capacity.(int) == 1:
+				case capacity.(int) == 2:
+				case capacity.(int) == 4:
+				case capacity.(int) == 8:
+				case capacity.(int) == 16:
+				case capacity.(int) == 24:
+				case capacity.(int) == 32:
+				case capacity.(int) == 40:
+				case capacity.(int) == 80:
+				default:
+					return fmt.Errorf("BusinessCritical pricing tier must have a capacity of 2, 4, 8, 16, 24, 32, 40, or 80 vCores")
+				}
+			}
+
+			// Addutional checks based of SKU type...
+			if strings.HasPrefix(strings.ToLower(name.(string)), "gp_") || strings.HasPrefix(strings.ToLower(name.(string)), "bc_") {
+				// vCore based
+				if maxCapacity.(float64) > float64(capacity.(int)) {
+					return fmt.Errorf("BusinessCritical and GeneralPurpose pricing tiers perDatabaseSettings maxCapacity must not be higher than the SKUs capacity value")
+				}
+
+				if minCapacity.(float64) > maxCapacity.(float64) {
+					return fmt.Errorf("perDatabaseSettings maxCapacity must be greater than or equal to the perDatabaseSettings minCapacity value")
+				}
+			} else {
+				// DTU based
+				if maxCapacity.(float64) != math.Trunc(maxCapacity.(float64)) {
+					return fmt.Errorf("BasicPool, StandardPool, and PremiumPool SKUs must have whole numbers as their maxCapacity")
+				}
+
+				if minCapacity.(float64) != math.Trunc(minCapacity.(float64)) {
+					return fmt.Errorf("BasicPool, StandardPool, and PremiumPool SKUs must have whole numbers as their minCapacity")
+				}
+
+				if minCapacity.(float64) < 0.0 {
+					return fmt.Errorf("BasicPool, StandardPool, and PremiumPool SKUs per_database_settings min_capacity must be equal to or greater than zero")
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func resourceArmMsSqlElasticPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).msSqlElasticPoolsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for MsSQL ElasticPool creation.")
+
+	elasticPoolName := d.Get("name").(string)
+	serverName := d.Get("server_name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	resGroup := d.Get("resource_group_name").(string)
+	sku := expandAzureRmMsSqlElasticPoolSku(d)
+	properties := expandAzureRmMsSqlElasticPoolProperties(d)
+	tags := d.Get("tags").(map[string]interface{})
+
+	elasticPool := sql.ElasticPool{
+		Sku:                   sku,
+		ElasticPoolProperties: properties,
+		Location:              &location,
+		Tags:                  expandTags(tags),
+		Name:                  &elasticPoolName,
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, serverName, elasticPoolName, elasticPool)
+	if err != nil {
+		return err
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
```

Comment:
Avoid the shadows declaration lint error by re-using the err object instead of declaring it a second time.

```suggestion
	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
```